File: dmn-test-cases/standard/tck/1.1/cl3/0013-sort/translator/expected/java/dmn/Sort2.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"decision.ftl", "sort2"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "sort2",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class Sort2 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "sort2",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
        com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    public Sort2() {
    }

    @java.lang.Override()
    public List<type.TRow> applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply((input_.get("tableB") != null ? com.gs.dmn.serialization.JsonSerializer.OBJECT_MAPPER.readValue(input_.get("tableB"), new com.fasterxml.jackson.core.type.TypeReference<List<type.TRow>>() {}) : null), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'Sort2'", e);
            return null;
        }
    }

    public List<type.TRow> apply(List<type.TRow> tableB, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start decision 'sort2'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long sort2StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments sort2Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            sort2Arguments_.put("tableB", tableB);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, sort2Arguments_);

            // Evaluate decision 'sort2'
            List<type.TRow> output_ = lambda.apply(tableB, context_);

            // End decision 'sort2'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, sort2Arguments_, output_, (System.currentTimeMillis() - sort2StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'sort2' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<List<type.TRow>> lambda =
        new com.gs.dmn.runtime.LambdaExpression<List<type.TRow>>() {
            public List<type.TRow> apply(Object... args_) {
                List<type.TRow> tableB = 0 < args_.length ? (List<type.TRow>) args_[0] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 1 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[1] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                return sort(tableB, new com.gs.dmn.runtime.LambdaExpression<Boolean>() {public Boolean apply(Object... args_) {type.TRow x = (type.TRow)args_[0]; type.TRow y = (type.TRow)args_[1];return numericLessThan(((java.lang.Number)(x != null ? x.getCol2() : null)), ((java.lang.Number)(y != null ? y.getCol2() : null)));}});
            }
        };
}

File: dmn-test-cases/standard/tck/1.1/cl3/0013-sort/translator/expected/kotlin/dmn/Sort2.kt

import java.util.*
import java.util.stream.Collectors

@javax.annotation.Generated(value = ["decision.ftl", "sort2"])
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "sort2",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
class Sort2() : com.gs.dmn.runtime.JavaTimeDMNBaseDecision<List<type.TRow?>?>() {
    override fun applyMap(input_: MutableMap<String, String>, context_: com.gs.dmn.runtime.ExecutionContext): List<type.TRow?>? {
        try {
            return apply(input_.get("tableB")?.let({ com.gs.dmn.serialization.JsonSerializer.OBJECT_MAPPER.readValue(it, object : com.fasterxml.jackson.core.type.TypeReference<List<type.TRow?>?>() {}) }), context_)
        } catch (e: Exception) {
            logError("Cannot apply decision 'Sort2'", e)
            return null
        }
    }

    fun apply(tableB: List<type.TRow?>?, context_: com.gs.dmn.runtime.ExecutionContext): List<type.TRow?>? {
        try {
            // Start decision 'sort2'
            var annotationSet_: com.gs.dmn.runtime.annotation.AnnotationSet = context_.getAnnotations()
            var eventListener_: com.gs.dmn.runtime.listener.EventListener = context_.getEventListener()
            var externalExecutor_: com.gs.dmn.runtime.external.ExternalFunctionExecutor = context_.getExternalFunctionExecutor()
            var cache_: com.gs.dmn.runtime.cache.Cache = context_.getCache()
            val sort2StartTime_ = System.currentTimeMillis()
            val sort2Arguments_ = com.gs.dmn.runtime.listener.Arguments()
            sort2Arguments_.put("tableB", tableB)
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, sort2Arguments_)

            // Evaluate decision 'sort2'
            val output_: List<type.TRow?>? = evaluate(tableB, context_)

            // End decision 'sort2'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, sort2Arguments_, output_, (System.currentTimeMillis() - sort2StartTime_))

            return output_
        } catch (e: Exception) {
            logError("Exception caught in 'sort2' evaluation", e)
            return null
        }
    }

    private inline fun evaluate(tableB: List<type.TRow?>?, context_: com.gs.dmn.runtime.ExecutionContext): List<type.TRow?>? {
        var annotationSet_: com.gs.dmn.runtime.annotation.AnnotationSet = context_.getAnnotations()
        var eventListener_: com.gs.dmn.runtime.listener.EventListener = context_.getEventListener()
        var externalExecutor_: com.gs.dmn.runtime.external.ExternalFunctionExecutor = context_.getExternalFunctionExecutor()
        var cache_: com.gs.dmn.runtime.cache.Cache = context_.getCache()
        return sort(tableB, com.gs.dmn.runtime.LambdaExpression<Boolean> { args_ -> val x: type.TRow? = args_[0] as type.TRow?; val y: type.TRow? = args_[1] as type.TRow?; numericLessThan(x?.let({ it.col2 as kotlin.Number? }), y?.let({ it.col2 as kotlin.Number? })) }) as List<type.TRow?>?
    }

    companion object {
        val DRG_ELEMENT_METADATA : com.gs.dmn.runtime.listener.DRGElement = com.gs.dmn.runtime.listener.DRGElement(
            "",
            "sort2",
            "",
            com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
            com.gs.dmn.runtime.annotation.ExpressionKind.LITERAL_EXPRESSION,
            com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
            -1
        )
    }
}
